old-ui/app/components/confirm-change-password.js

```diff
@@ -0,0 +1,149 @@
+    warning: state.appState.warning,
+  }
+}
+
+inherits(ConfirmChangePassword, Component)
+function ConfirmChangePassword () {
+  Component.call(this)
+}
+
+ConfirmChangePassword.prototype.render = function () {
+  const state = this.props
+  const passwordInputAdditionalStyle = {
+    width: '100%',
+    marginTop: 10,
+    marginBottom: 20,
+  }
+  return h('.flex-column.flex-grow', {
+    style: {
+      overflowX: 'auto',
+      overflowY: 'hidden',
+    },
+  }, [
+    // subtitle and nav
+    h('.section-title.flex-row.flex-center', [
+      h('i.fa.fa-arrow-left.fa-lg.cursor-pointer', {
+        onClick: () => {
+          this.props.dispatch(actions.showConfigPage())
+        },
+        style: {
+          position: 'absolute',
+          left: '30px',
+        },
+      }),
+      h('h2.page-subtitle', 'Change Password'),
+    ]),
+    h('div', {
+      style: {
+        margin: '0 30px',
+      },
+    }, [
+      h('.error', {
+        style: {
+          display: state.warning ? 'block' : 'none',
+        },
+      }, state.warning),
+      h('span', 'Old password'),
+      h('input.large-input', {
+        type: 'password',
+        id: 'old-password-box',
+        style: passwordInputAdditionalStyle,
+      }),
+      h('span', 'New password'),
+      h('input.large-input', {
+        type: 'password',
+        id: 'new-password-box',
+        style: passwordInputAdditionalStyle,
+      }),
+      h('span', 'Confirm new password'),
+      h('input.large-input', {
+        type: 'password',
+        id: 'password-box-confirm',
+        style: passwordInputAdditionalStyle,
+        onKeyPress: this.createOnEnter.bind(this),
+      }),
+    ]),
+    h('p.confirm-label', {
+        style: {
+          textAlign: 'center',
+          margin: '0px 30px 20px ',
+        },
+      },
+      `Are you sure you want to change the password for unlocking of your wallet?`),
+    h('.flex-row.flex-right', {
+      style: {
+        marginRight: '30px',
+      },
+    }, [
+      h('button.btn-violet',
+        {
+          style: {
+            marginRight: '10px',
+          },
+          onClick: () => {
+            this.props.dispatch(actions.showConfigPage())
+          },
+        },
+        'No'),
+      h('button',
+        {
+          onClick: () => {
+            this.ChangePassword()
+          },
+        },
+        'Yes'),
+    ]),
+  ])
+}
+
+ConfirmChangePassword.prototype.createOnEnter = function (event) {
+  if (event.key === 'Enter') {
+    this.ChangePassword()
+  }
+}
+
+ConfirmChangePassword.prototype.ChangePassword = function () {
+  var oldPasswordBox = document.getElementById('old-password-box')
+  var oldPassword = oldPasswordBox.value
+  var newPasswordBox = document.getElementById('new-password-box')
+  var newPassword = newPasswordBox.value
+  var newPasswordConfirmBox = document.getElementById('password-box-confirm')
+  var newPasswordConfirm = newPasswordConfirmBox.value
+  if (oldPassword.length < 8) {
```

Comment:
If for some reason a user creates a password with less than 8 characters, this won't allow changing it. I know that passwords are validated to have that minimum length, but you never know. I would remove this unless there's some good reason to keep it.

Reply:
Ok, I skipped this validation here. Indeed, it looks redundant here.